File: node/src/mailing.js
const axios = require("axios");

class Mailing {
  base_url = "https://mailing.go-mailer.com";

  constructor(api_key) {
    this.api_key = api_key;
  }

  async sendTransactionalEmail({
    template_code,
    recipient_email,
    data = {},
    html_markup = "",
    bcc = "",
    attachments = [],
  }) {
    if (!template_code) throw new Error("Event code is required.");
    if (!recipient_email) throw new Error("Recipient email is required.");

    const body = { template_code, recipient_email, data, html: html_markup, bcc, attachments };
    const { data: response } = await axios.post(`${this.base_url}/api/v1/transactionals/dispatch`, body, {
      headers: { Authorization: `Bearer ${this.api_key}` },
    });

    return response;
  }
}

module.exports = { Mailing };

File: python/gomailer/mailing.py
import requests


class Mailing:
    base_url = 'https://api.go-mailer.com'

    def __init__(self, api_key):
        self.api_key = api_key

    def send_transactional_email(self, template_code, recipient_email, data={}, html_markup='', attachments=[], bcc=''):
        if not template_code:
            raise Exception('Template code is required.')
        elif not recipient_email:
            raise Exception('Recipient email is required')

        body = {
            'template_code': template_code,
            'recipient_email': recipient_email,
            'data': data,
            'html': html_markup,
            'attachments': attachments,
            'bcc': bcc
        }

        headers = {'Authorization': f'Bearer {self.api_key}'}

        response = requests.post(
            f"{self.base_url}/v1/transactionals", json=body, headers=headers)
        return response
